cilium/CFP-39876-clustermesh-filtered-export.md

```diff
@@ -0,0 +1,211 @@
+- Cross-cluster pod to pod connectivity inside global namespaces
+- Global service and MCS-API support for resources under global namespaces
+
+## Non-Goals
+
+- Network policies for non global(local) namespaces
+- Cross-cluster pod to pod connectivity for non global(local) namespaces
+- Global service and MCS-API support for non global(local) namespaces
+- Support Global/Local namespaces for non CRD modes of identity allocation 
+ 
+## Network Policy 
+### Cross-Cluster Communication and Network Policy Behavior
+
+Full support for **cross-cluster communication** and **network policy enforcement** is available **only when both the source (client) and destination (server) pods**—which may optionally be accessed via a service—**are part of a _global namespace_**. Network policies are not enforced if either the source or destination pod is in a **non-global namespace**.
+
+### Behavior When Pods Are Not in Global Namespaces
+Network policies in Cilium rely heavily on **identity labels** for policy matching. Cross-cluster traffic from a source pod not part of a global namespace to a destination pod matched by an ingress policy is dropped, unless from-world traffic is explicitly allowed by the policy, in which case it may be allowed, depending on the specific Cilium configuration. Similarly, cross-cluster traffic from a source pod matched by an egress policy to a destination pod not part of a global namespace is dropped, unless to-world traffic is explicitly allowed by the policy (either globally, or via a specific CIDR/FQDN rule), in which case it is allowed.
+
+#### Why Current Policies Can't Work
+
+Cilium `NetworkPolicy` typically uses selectors like:
+
+```yaml
+endpointSelector:
+  matchLabels:
+    app: frontend
+ingress:
+  - fromEndpoints:
+    - matchLabels:
+        app: backend
+        tier: api
+```
+
+Without labels, the policy engine cannot:
+-Match fromEndpoints selectors
+-Evaluate matchLabels conditions
+-Distinguish between different application identities
+
+
+When using this feature, cross-cluster communication requires both the source and destination namespaces to be global, and both clusters to have an established cluster mesh. The evaluation of network policy for different cases is summarized below:
+
+### Summary of Enforcement Logic
+
+| Source Pod in Global NS | Destination Pod in Global NS | Can you allow traffic based on Labels? |
+|-------------------------|------------------------------|------------------|
+| Yes                     | Yes                          | ✅ Enforced normally |
+| No                      | Yes                          | ❌ Not Supported  |
+| Yes                     | No                           | ❌ Not Supported |
+| No                      | No                           | ❌ Not Supported  |
+
+In the above table, Not Supported implies the policies are not enforced due to missing source identity or labels information at the destination cluster. 
```

Comment:
Wording around "policies are not enforced" can be misinterpreted to imply that network policy enforcement will be completely disabled, but that is not what the CFP proposes.

```suggestion
In the above table, Not Supported implies that policy enforcement will not take into account labels on Pods in local namespaces in a peer cluster. 
```

We might clarify further that this means that even if a policy explicitly allows traffic for a remote cluster for pods with specific labels, that policy may not allow the traffic if the remote Pod's namespace is local. 

I think we're on the same page here, but there is a subtlety in the wording between "policy enforcement has surprising behavior" vs. "there is no policy enforcement".